### PR TITLE
Guessit 3.0

### PR DESCRIPTION
**Pull request by sarendsen**
### Motivation for changes:
Upgrade to guessit 3.0 so we can take advantage of new features and bugfixes.

### Detailed changes:
- Rewrite initiation of GuessItApi for 3.0
- Added mapping for new guessit namings
- Bump requirements versions

### Implemented feature requests:
- https://github.com/Flexget/Flexget/issues/2160

### CODE REVIEW

**Comment by sarendsen** [comment]
So, there is one unit test failing. See https://circleci.com/gh/Flexget/Flexget/4569
Which seems to be a bug in guessit. It doesnt parse the episode range correctly on that one.
Wondering howto deal whit this. Remove it or make work of it (meaning putting in a ticket at the guessit repo and waiting for a new release)

Fun fact. It works when not giving an expected_title to guessit. 

```
guessit 'Something.4x05-06'
For: Something.4x05-06
GuessIt found: {
    "title": "Something",
    "season": 4,
    "episode": [
        5,
        6
    ],
    "type": "episode"
}
```
but fails with expected title 
```
guessit 'Something.4x05-06' -T Something
For: Something.4x05-06
GuessIt found: {
    "title": "Something",
    "season": 4,
    "episode": 5,
    "release_group": "06",
    "type": "episode"
}
```

**Comment by cvium** [comment]
I tried upgrading a few months ago and I ran into the same problem. I just gave up. Last time I opened a few issues on their github repo and they were quick to fix them, but their release schedule is slow.

**Comment by sarendsen** [comment]
Exactly, even if we get this fixed on there side we will be waiting for a release. As this only breaks one unit test, might the maintainers consider this to be a tradeoff? Im open for ideas

**Comment by JohnDoee** [comment]
GuessIt development is basically dead so waiting for it to get fixed might not be viable.

**Comment by snickers2k** [comment]
> GuessIt development is basically dead so waiting for it to get fixed might not be viable.

Why you think that? 3.0.0 is released on  2018-05-22 and there is a Version 3.0.1 in developement. But since we are still using 2.1.4  it would be great to have flexget support for 3.0

**Comment by cvium** [comment]
> we are still using 2.1.4 it would be great to have flexget support for 3.0

The bug (https://github.com/Flexget/Flexget/pull/2197#issuecomment-416686350) needs to be fixed before that can happen.

**Comment by gazpachoking** [comment]
Did someone open an issue at guessit? Hard for them to fix it, or for us to track it without that.

**Comment by snickers2k** [comment]
i just did that with referring to this issue here https://github.com/guessit-io/guessit/issues/577
we'll see if they even take a look at it. Maybe JohnDoee was right with guessit developement being dead. I Don't know.  
I guess i'll just stick with internal for now and see if guessit goes forward anytime. For now, i think sarendsen is right - guessit maybe should be just removed at this state and point. Maybe with just linking the guessit string to internal, so most people don't  have to change anything. but thats on you.

**Comment by Toilal** [comment]
Hey, guessit development is not dead 😊, I just released a new release 3.0.2 and will try to have a look to this use case in the next few days. Stay tuned.

**Comment by Toilal** [comment]
guessit has just been released with the fix (`3.0.3`).

Note that [series keyword has been removed](https://github.com/guessit-io/guessit/commit/12dc120e2c3a8f3114c190053d165bbede1808a0#diff-eba58ec782971c34eb1c1b92476130f0L156) from the default configuration, so [FlexGet.Series.2013.14.of.21.Title.Here.720p.HDTV.AAC5.1.x264-NOGRP use case](https://github.com/Flexget/Flexget/blob/80fdc98886c70082de5efab699f1d4afa4f82870/flexget/tests/test_seriesparser.py#L605) will probably fail ... 

This keyword was removed because of [this](https://github.com/guessit-io/guessit/issues/550) and [this](https://github.com/guessit-io/guessit/issues/561#issuecomment-417865808).

You can still apply an [advanced configuration](https://github.com/guessit-io/guessit/blob/develop/docs/configuration.rst) to bring back this keyword in guessit, but i'll vote against it.

**Comment by sarendsen** [comment]
Updated to 3.0.3, and worked out some kinks. Ready for review i guess.

File: flexget/plugins/parsers/parser_guessit.py [hunk 1]
```diff
@@ -229,7 +263,7 @@ def parse_series(self, data, **kwargs):
             episode = episode[0]
         date = guess_result.get('date')
         quality = self._quality(guess_result)
-        proper_count = self._proper_count(guess_result)
+        proper_count = self._proper_count(guess_result, )
```
Comment (cvium) [inline flexget/plugins/parsers/parser_guessit.py]: Why the comma?

Comment (sarendsen) [reply flexget/plugins/parsers/parser_guessit.py]: Nice catch. Small typo. Committed a fix.

**Comment by sarendsen** [comment]
@cvium Bump :)

**Comment by snickers2k** [comment]
+1
too many errors with 2.x :(

**Comment by cvium** [comment]
Seems like there's a lot of errors with 3.x as well. 

https://github.com/Flexget/Flexget/issues/2249
https://discuss.flexget.com/t/unhandled-error-in-lazy-lookup-plugin-unhashable-type-list/4384/

**Comment by Toilal** [comment]
There are options to enforce single value or list.

```
-s, --single-value Keep only first value found for each property
-l, --enforce-list Wrap each found value in a list even when property has
a single value
```


Le mar. 13 nov. 2018 à 07:25, Rémi Alvergnat <toilal.dev@gmail.com> a
écrit :

**Comment by sarendsen** [comment]
@Toilal: Thanks for your taste response
@cvium: Based on the log and Toilal's response this should be fixable on our side. Ill see if i can put out a fix today.

**Pull request merged**